ledger/cryptarchia-ledger/src/lib.rs

```diff
@@ -25,9 +26,9 @@ pub enum LedgerError {
     #[error("Invalid block slot {block:?} for parent slot {parent:?}")]
     InvalidSlot { parent: Slot, block: Slot },
     #[error("Parent block not found: {0:?}")]
-    ParentNotFound(HeaderId),
+    ParentNotFound(Id),
     #[error("Orphan block missing: {0:?}. Importing leader proofs requires the block to be validated first")]
-    OrphanMissing(HeaderId),
+    OrphanMissing(Id),
```

Comment:
Do we need to include the Id in these error variants? Do you use the id when handling these errors? (I'm personally not a fan of Error enums with generics)

Reply:
It makes it a bit easier to handle errors. How do you know which of the imported orphans was the cause of problems? What would you suggest?

Reply:
I guess my question is, do you actually need to know which orphan was the cause of the issue? If not, I'd wait till you actually need to before adding it here

The generics on errors blow up the method type signatures and causes more of them to wrap and you end up having to add function type paramaters where you wouldn't otherwise.

Reply:
>  do you actually need to know which orphan was the cause of the issue? 
 
  A bunch of things coming to my mind:
  * debug prints ( I guess you could make them inside the ledger, but not a big fan of this)
  * requesting missing block
  * checking in tests that the error is indeed that one you expect
  
 > The generics on errors blow up the method type signatures and causes more of them to wrap and you end up having to add function type paramaters where you wouldn't otherwise.
 
 True, but I would expect most of services to just use `HeaderId` here, (and maybe even export a type alias for convenience)